src/sensor/ping360.cpp

```diff
@@ -151,6 +145,32 @@ void Ping360::handleMessage(const ping_message& msg)
         break;
     }
 
+    case CommonId::NACK: {
+        const common_nack nack(msg);
+        if (nack.nacked_id() == Ping360Id::TRANSDUCER) {
+            qCWarning(PING_PROTOCOL_PING360) << "transducer control was NACKED, reverting to default settings";
+
+            _gain_setting = _firmwareDefaultGainSetting;
+            _transmit_duration = _firmwareDefaultTransmitDuration;
+            _sample_period = _firmwareDefaultSamplePeriod;
+            _transmit_frequency = _firmwareDefaultTransmitFrequency;
+            _num_points = _firmwareDefaultNumberOfSamples;
+
+            // request another transmission
+            requestNextProfile();
+
+            // restart timer
+            _timeoutProfileMessage.start();
+
+            emit gainSettingChanged();
+            emit transmitDurationChanged();
+            emit samplePeriodChanged();
+            emit transmitFrequencyChanged();
+            emit numberOfPointsChanged();
```

Comment:
`emit rangeChanged();` here to solves the issue I have messing with the transmit duration.

Reply:
I know ;)